test/e2e/specs/editor/blocks/navigation.spec.js

```diff
@@ -802,6 +802,500 @@ test.describe( 'Navigation block', () => {
 	} );
 } );
 
+test.describe( 'Navigation block - Frontend interactivity', () => {
+	test.beforeAll( async ( { requestUtils } ) => {
+		await requestUtils.activateTheme( 'emptytheme' );
+		await requestUtils.deleteAllTemplates( 'wp_template_part' );
+	} );
+
+	test.beforeEach( async ( { requestUtils } ) => {
+		await Promise.all( [ requestUtils.deleteAllMenus() ] );
+	} );
+
+	test.afterAll( async ( { requestUtils } ) => {
+		await Promise.all( [
+			requestUtils.deleteAllMenus(),
+			requestUtils.activateTheme( 'twentytwentyone' ),
+		] );
+	} );
+
+	test.afterEach( async ( { requestUtils } ) => {
+		await requestUtils.deleteAllPosts();
+		await requestUtils.deleteAllMenus();
+		await requestUtils.deleteAllTemplates( 'wp_template_part' );
+	} );
+
+	test.describe( 'Overlay menu', () => {
+		test.beforeEach( async ( { admin, editor, page, requestUtils } ) => {
```

Comment:
This could be done once for all subtests:
```suggestion
		test.beforeAll( async ( { admin, editor, page, requestUtils } ) => {
```

Reply:
It seems I cannot use `admin`, `editor` or `page` inside `beforeAll`. I'm getting this error:

```
Error: “context” and “page” fixtures are not supported in “beforeAll” since they are created on a per-test basis.
    If you would like to reuse a single page between tests, create context manually with browser.newContext(). See https://aka.ms/playwright/reuse-page for details.
    If you would like to configure your page before each test, do that in beforeEach hook instead.
```

Even if I don't use `page` (which is not needed), I keep getting the same error for `admin` and `editor`.

Reply:
Yep, some fixtures (`page`, `admin`, `editor`, for instance) can't be used inside `beforeAll` or `afterAll`. It's a limitation (feature?) of Playwright to enable running tests in isolation. Putting them inside `beforeEach` instead is generally a good alternative 👍 .

If the testing duration is a concern, I recommend grouping similar tests with the same setup into [longer ones](https://playwright.dev/docs/best-practices#write-fewer-tests-but-longer-tests) instead.

> Even if I don't use `page` (which is not needed), I keep getting the same error for `admin` and `editor`.

That's because both fixtures use `page` internally.

Reply:
`beforeEach` it is then 😄